src/main/java/gg/solarmc/clans/command/commands/clans/KickCommand.java

```diff
@@ -0,0 +1,80 @@
+package gg.solarmc.clans.command.commands.clans;
+
+import gg.solarmc.clans.command.SubCommand;
+import gg.solarmc.clans.helper.PluginHelper;
+import gg.solarmc.loader.DataCenter;
+import gg.solarmc.loader.SolarPlayer;
+import gg.solarmc.loader.clans.Clan;
+import gg.solarmc.loader.clans.ClanMember;
+import gg.solarmc.loader.clans.ClansKey;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.bukkit.ChatColor;
+import org.bukkit.Server;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+public class KickCommand implements SubCommand {
+    @Override
+    public void execute(CommandSender sender, String[] args, PluginHelper helper) {
+        if (helper.invalidateCommandSender(sender, args)) return;
+        Player player = (Player) sender;
+        if (helper.invalidateArgs(sender, args,
+                ChatColor.RED + "You need to specify the Name of the Player you want to Invite!!")) return;
+
+        Clan clan = player.getSolarPlayer().getData(ClansKey.INSTANCE).currentClan().orElse(null);
+
+        if (clan == null) {
+            helper.sendNotInClanMsg(player);
+            return;
+        }
+
+        if (!helper.isLeader(clan, player)) {
+            player.sendMessage(Component.text("Only Clan Leader can use this Command", NamedTextColor.RED));
+            return;
+        }
+
+        Server server = player.getServer();
+        DataCenter dataCenter = server.getDataCenter();
+
+        dataCenter.runTransact(
+                transaction -> {
+                    SolarPlayer solarPlayerKicked = dataCenter.lookupPlayerUsing(transaction, args[0]).orElse(null);
+
+                    if (solarPlayerKicked == null) {
+                        player.sendMessage(ChatColor.RED + "Cannot find the player!");
+                        return;
+                    }
+
+                    if (clan.currentMembers().contains(new ClanMember(solarPlayerKicked.getUserId()))) {
+                        player.sendMessage(ChatColor.RED + "Player is not in your clan!!");
+                        return;
+                    }
+
+                    Player playerKicked = server.getPlayer(solarPlayerKicked.getMcUuid());
```

Comment:
There's a structural issue here. A `SolarPlayer` does not need to be online, in contrast to an `OnlineSolarPlayer`. Player.getSolarPlayer() returns an OnlineSolarPlayer, whereas `lookupPlayerUsing` returns a SolarPlayer because it can find offline users.

Meanwhile, `server.getPlayer` returns online players only. If your objective is to find online players, use `server.getPlayer` then Player#getSolarPlayer.

Also, server.getPlayer is not thread-safe, so you should not put it inside a transaction run.

Reply:
How can i get the name of the offline player ? I just need the player's name